eodag/rest/utils.py

```diff
@@ -579,9 +592,42 @@ def download_stac_item_by_id_stream(catalogs, item_id, provider=None):
     :returns: a stream of the downloaded data (either as a zip or the individual assets)
     :rtype: StreamingResponse
     """
-    product = search_product_by_id(
-        item_id, product_type=catalogs[0], provider=provider
-    )[0]
+    product_type = catalogs[0]
+    search_plugin = next(
+        eodag_api._plugins_manager.get_search_plugins(product_type, provider)
+    )
+    provider_product_type_config = search_plugin.config.products.get(
+        product_type, {}
+    ) or search_plugin.config.products.get(GENERIC_PRODUCT_TYPE, {})
+    if provider_product_type_config.get("storeDownloadUrl", False):
+        if item_id not in search_plugin.download_info:
+            logger.error(f"data for item {item_id} not found")
+            raise IdNotFoundError(
+                f"download url for product {item_id} could not be found, please redo "
+                f"the search request to fetch the required data"
+            )
+        product_data = search_plugin.download_info[item_id]
+        properties = {
+            "id": item_id,
+            "orderLink": product_data["orderLink"],
+            "downloadLink": product_data["downloadLink"],
+            "geometry": "-180 -90 180 90",
+        }
+        product = EOProduct(provider or product_data["provider"], properties)
+    else:
+        products = search_product_by_id(
+            item_id, product_type=catalogs[0], provider=provider
+        )
+        if not products:
+            logger.info(
+                f"No product found matching id {item_id} for product_type {catalogs[0]}"
+            )
+            raise HTTPException(
```

Comment:
use NotAvailableError, that is then handled from server.py

Reply:
fixed